lib/bolt/pal/yaml_plan/evaluator.rb

```diff
@@ -17,12 +17,9 @@ def initialize(analytics = Bolt::Analytics::NoopClient.new)
         def dispatch_step(scope, step)
           step = evaluate_code_blocks(scope, step)
 
-          step_type, *extra_keys = STEP_KEYS.select { |key| step.key?(key) }
-          if !step_type || extra_keys.any?
-            unsupported_step(scope, step)
-          end
+          step_type = STEP_KEYS.select { |key| step.key?(key) }
```

Comment:
`.select.first` => `.find`

Reply:
Updated. I remember thinking that looked clunky. Never got around to investigating. Thanks for enlightening me.